### PR TITLE
Add `useless-object-inheritence` rule for pylint 

### LINKED ISSUE #6269: Enable `useless-object-inheritance` for pylint
### Task

Enable `useless-object-inheritance` for pylint to prevent useless object inheritance:
https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/useless-object-inheritance.html

### Lines that need to be changed

https://github.com/mlflow/mlflow/blob/4f5d3f095da9ed6699941d8aacb9925488ff2fb5/pylintrc#L174-L177

### PR DESCRIPTION
**Pull request by KarthikKothareddy**
Signed-off-by: Karthik Kothareddy <karthikreddy.kp@gmail.com>

## Related Issues/PRs

Close #6269 

## What changes are proposed in this pull request?

Changes to include `useless-object-inheritence` rule for pylint.

## How is this patch tested?

- [x] I have written tests (not required for typo or doc fix) and confirmed the proposed feature/bug-fix/change works.

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/pipelines`: Pipelines, Pipeline APIs, Pipeline configs, Pipeline Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by harupy — APPROVED** [review]
LGTM!

File: pylintrc [hunk 1]
```diff
@@ -173,7 +173,8 @@ disable=print-statement,
 # it should appear only once). See also the "--disable" option for examples.
 enable=c-extension-no-member,
        line-too-long,
-       use-a-generator
+       use-a-generator,
+       useless-object-inheritance
```
Comment (harupy) [inline pylintrc:177]: I think this change will conflict with https://github.com/mlflow/mlflow/pull/6321. Can you resolve conflicts once https://github.com/mlflow/mlflow/pull/6321 is merged?

Comment (KarthikKothareddy) [reply pylintrc:177]: Of course.

**Review by dbczumar — APPROVED** [review]
@KarthikKothareddy This looks great! Thank you for your contribution! Before merging, can you sign your commits so that our DCO check passes? You can find instructions here: https://github.com/mlflow/mlflow/pull/6322/checks?check_run_id=7505100364

**Comment by harupy** [comment]
@KarthikKothareddy 

Can you run this command to fix the DCO check failure?

```
git rebase HEAD~7 --signoff
Fgit push --force-with-lease origin karthik-fix-6269
```

**Comment by harupy** [comment]
Manually signed off https://github.com/mlflow/mlflow/pull/6322/commits/05953d7d74b912f19401ecc342fbdaa5863c7c7e when merging the PR:

image

**Comment by KarthikKothareddy** [comment]
> @KarthikKothareddy
> 
> I manually signed off [05953d7](https://github.com/mlflow/mlflow/commit/05953d7d74b912f19401ecc342fbdaa5863c7c7e) when merging the PR:
> 
> image
@harupy and @dbczumar my bad, I didn't had access to computer after I pushed my changes for a minute and thanks for Signing it off.

**Pull request merged**